src/transaction/guards.ts

```diff
@@ -16,14 +16,18 @@ import {
   TxBodyKeys,
 } from './types'
 
-export const isLovelace = (test: any): test is number => {
+export const isBigInt = (test: any): test is BigInt => typeof test === 'bigint'
```

Comment:
Do we really need this guard? I would probably just change the `isBigNumber` function so it checks `if (typeof test === 'number' || typeof test === 'bigint') return true`. I would also probably rename it to `isNumber` since it guards that. What do you think? Also it makes a bit more sence to me to not explictly check whether something is a BigNumber, but only to check whether it is a number. (for example in the current usage in the cryptoproviders).

Reply:
`isBigNumber` guard tests for `BigInt`, `isBigNumber` tests for `number`. If you want to use tested value as `BigInt` type, we need need the `test is BigInt` part. We are using `isBigNumber` before parsing to be sure there isn't string or object, it makes sense to keep `test is number` for this one. Merging these two functions would add code elsewhere or at least you would be losing (or getting wrong) type information.

Reply:
As mentioned, ttl and validityInterval shoulb be just numbers, not bigints, so we dont need this guard and in places we are currently using it we should use `isLovelace`. In that case I am not sure if we need the `isBigNumber` chceck as well and shouldnt remove that one as well (rename it back to `isLovelace`)

Reply:
Refactored